### DISCUSSION TITLE
利用方法、設置方法質問（この discussions は古いです。新しい discussions は #72 ）

### DISCUSSION THREAD

**Discussion by dounokouno** [dounokouno/TransmitMail#69] [Q&A]
GitHub で Discussions 機能が使えるようになったので、利用方法や設置方法等に不明点などあれば、こちらにコメントをお願いします 🙇‍♂️

以前の「バグ、不具合報告」issue は → #2

# 2021年3月10日 追記

issue #2 を利用していた「利用方法、設置方法質問」が discussions にコンバートできたので、以降はコンバートできた先の「利用方法、設置方法質問」discussions をご利用くださいmm

discussion #72

**Comment by 7st-wa3** [comment]
dounokouno 様

こんにちは。お世話になります。
TransmitMailをいつも利用させていただいております。
今回、以下のような使い方はできないかと、方法を模索しております。

1. １セットのTransmitMailを複数の入り口から使う
2. 入り口から入る際、getかpostかで「メールの送り先」と「メールのタイトル」を指定する
3. フォームの入力項目自体は同じで、2で指定した送り先に指定したメールタイトルで送信する

要するに「メールの送り先」と「メールのタイトル」を動的に、ページ側で指定したいわけです。
こういった指定（プログラムの改造？）は可能でしょうか。

お忙しいところ申し訳ございませんが、アドバイスいただけましたら幸いです。

**Reply by dounokouno** [reply]
@7st-wa3 様

コメントいただいた件ですが、例えば、メールフォームの URL を下記のように2つ用意する、という形でも問題ないでしょうか？

- https://www.example.com/contacts/?category=1
- https://www.example.com/contacts/?category=2

`category` と `1` や `2` は、実際にはもっとわかりやすいものに変えていただいて大丈夫です。下記のように値は無しでもいいかもしれません。

- https://www.example.com/contacts/?category1
- https://www.example.com/contacts/?category2

上記の例ですと、 PHP 側で `GET['category1']` が存在するかしないか、または `GET['category2']` が存在するかしないかで、入り口を判別することが可能です。（前者の場合ですと `if (GET['category'] === '1') {}` または `if (GET['category'] === '2') {}` で判別します。）

上記の例で入り口を判別いただいて、それプラス、下記は私が以前書いた Qiita の記事ですが、下記では `select` で選択された内容によってメール送信先を切り替えていますが、前述の方法で入り口を判別して、それを元にメール送信先やメールの件名を切り替えていただく形で対応可能かと思います。

[TransmitMail 2 で入力内容に応じて送信メールの宛先を切り替える - Qiita](https://qiita.com/dounokouno/items/21dcf87f2e43a73be551)

上記のような仕様で良くて、もう少し具体的なコードのサンプルが必要でしたら、もう少し具体的なコードのサンプルを書いてみたいと思います。

**Reply by 7st-wa3** [reply]
dounokouno 様

早速のご回答ありがとうございました！
ご提案いただいた方法で試してみます！
うまく行かなかったらまた相談させていただきます。

**Reply by 7st-wa3** [reply]
dounokouno 様

すいません、追加でもう一つ質問があります。

引数でメール送信先や件名を変更は、お知らせいただいたカスタマイズでできそうなんですが、
それに加えて自動返信メールの内容を切り替えることは可能でしょうか。

「mail_auto_reply_body.txt」を違う名前で複数用意して引数によって切り替えるようにするのか・・・
あるいは「mail_auto_reply_body.txt」の中で引数によって分岐させるような書き方をするのか・・・

正直プログラムのことがまだよくわかっていないところが多いのでうまく説明できませんが（汗）

申し訳ありませんがアドバイスよろしくおねがいします。

**Reply by dounokouno** [reply]
@7st-wa3 様

> 引数でメール送信先や件名を変更は、お知らせいただいたカスタマイズでできそうなんですが、
> それに加えて自動返信メールの内容を切り替えることは可能でしょうか。

はい、可能です。

メール本文の内容が全く違うのであれば、「mail_auto_reply_body.txt」とは別の「mail_auto_reply_body02.txt」といった名前のファイルを作ってもらって、 YAML ファイルであれば、下記のように書いてもらって、

```yaml
config:
    # 下記は初期値なので書かなくて OK
    mail_auto_reply_body: config/mail_auto_reply_body.txt
    # mail_auto_reply_body01 の部分はわかりやすい名前に設定してください
    mail_auto_reply_body02: config/mail_auto_reply_body02.txt
```

[lib/TransmitMail.php の1114行目あたり](https://github.com/dounokouno/TransmitMail/blob/master/lib/TransmitMail.php#L1114)を下記のように修正することで対応できるかと思います。

```php
// メール本文
$body = '';

if (...) {
    $body = $this->tpl->fetch($this->config['mail_auto_reply_body02']);
} else {
    $body = $this->tpl->fetch($this->config['mail_auto_reply_body']);
}

$body = $this->hd($body);
```

上記コードは私の方では検証していませんので、あらかじめご了承ください。

メール内容の一部を書き換えたい場合は、テンプレート記法の `if` が利用できるので、そちらで対応いただいた方が早いかもしれません。

[テンプレートの記述方法 · dounokouno/TransmitMail Wiki](https://github.com/dounokouno/TransmitMail/wiki/%E3%83%86%E3%83%B3%E3%83%97%E3%83%AC%E3%83%BC%E3%83%88%E3%81%AE%E8%A8%98%E8%BF%B0%E6%96%B9%E6%B3%95)

**Reply by 7st-wa3** [reply]
dounokouno 様

ご回答ありがとうございます！
試してみます！！

**Reply by 7st-wa3** [reply]
dounokouno 様

すいません、もう一つおしえてください！

[TransmitMail 2 で入力内容に応じて送信メールの宛先を切り替える - Qiita](https://qiita.com/dounokouno/items/21dcf87f2e43a73be551)
↑
の方法で、メールアドレスを変更しました。
その場合で、自動返信メールの中に、変更された送り先メアドを（署名等に）記載したいのですが、
可能でしょうか。その場合yamlでどう書いたらいいか教えていただきたく存じます。

**Reply by dounokouno** [reply]
@7st-wa3 様

例えば [lib/TransmitMail.php の975行目あたり](https://github.com/dounokouno/TransmitMail/blob/master/lib/TransmitMail.php#L975)にテンプレートファイルで参照できるようにするコードが書かれていますので、この辺りにコードを書き加えていただいても良いですし、 `$this->tpl->fetch($this->config['mail_auto_reply_body'])` より前であればどこで書き加えていただいても大丈夫かと思います。

例えば YAML が下記のようになっている場合、

```yml
config:
    email: admin@example.com
    email02: admin02@example.com
```

追加するコードは下記のようになり、

```php
$this->tpl->set('email02', $this->config['email02']);
```

テンプレートファイルでの呼び出しは下記のようになります。

```
株式会社テスト
http://www.example.com/
Email: {$email02}
TEL: 000-0000-0000
FAX: 000-0000-0000
```

上記コードは私の方では検証していませんので、あらかじめご了承ください。

**Reply by 7st-wa3** [reply]
dounokouno 様

おかげさまで希望の挙動にできました！
ありがとうございました！

**Comment by dounokouno** [comment]
issue #2 を利用していた「利用方法、設置方法質問」が discussions にコンバートできたので、以降はコンバートできた先の「利用方法、設置方法質問」discussions をご利用くださいmm

discussion #72
